cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -169,8 +159,10 @@ protected static String nextAndEnd(Iterator<String> iterator, String fieldName)
     return result;
   }
 
-  protected static String remaining(Iterator<String> iterator, String fieldName) {
-    Preconditions.checkArgument(iterator.hasNext(), "Missing field: %s", fieldName);
+  protected static String remaining(Iterator<String> iterator, @Nullable String fieldName) {
+    if (fieldName != null && !iterator.hasNext()) {
+      throw new IllegalArgumentException("Missing field: " + fieldName);
+    }
     StringBuilder result = new StringBuilder();
     while (iterator.hasNext()) {
       result.append(iterator.next()).append(IDSTRING_PART_SEPARATOR);
```

Comment:
Same logic as above. We can avoid the if check if the string builder begins with the `IDSTRING_PART_SEPARATOR`.